sys/arduino/include/spiport.hpp

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (C) 2019 Otto-von-Guericke-Universität Magdeburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_arduino_api
+ * @{
+ *
+ * @file
+ * @brief       Definition of the Arduino 'SPI' interface
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ */
+
+#ifndef SPIPORT_H
+#define SPIPORT_H
+
+#include "arduino_board.h"
+#include "byteorder.h"
+#include "periph/spi.h"
+#include "rmutex.h"
+
+/**
+ * @name    Arduino compatible SPI modes
+ * @{
+ */
+#define SPI_MODE0       (0)     /**< CPOL=0, CPHA=0 */
+#define SPI_MODE1       (1)     /**< CPOL=0, CPHA=1 */
+#define SPI_MODE2       (2)     /**< CPOL=1, CPHA=0 */
+#define SPI_MODE3       (3)     /**< CPOL=1, CPHA=1 */
+/** @} */
+
+/**
+ * @name    Arduino compatible SPI frequency selection via clock divider
+ *
+ * This API assumes the library was targeting a 16 MHz Arduino. It will choose
+ * the SPI clock frequency matching the requested frequency best, but never
+ * a frequency greater than the one it would have on Arduinos.
+ * @{
+ */
+#define SPI_CLOCK_DIV2      (0) /**< Best match for 8 MHz: 5 MHz */
+#define SPI_CLOCK_DIV4      (1) /**< Best match for 4 MHz: 1 MHz */
+#define SPI_CLOCK_DIV8      (1) /**< Best match for 2 MHz: 1 MHz */
+#define SPI_CLOCK_DIV16     (1) /**< Best match for 1 MHz: 1 MHz */
+#define SPI_CLOCK_DIV32     (2) /**< Best match for 500 kHz: 400 kHz */
+#define SPI_CLOCK_DIV64     (3) /**< Best match for 250 kHZ: 100 kHz */
+#define SPI_CLOCK_DIV128    (3) /**< Best match for 125 kHz: 100 kHz */
+/** @} */
+
+/**
+ * @name    Arduino compatible bit order values for SPI
+ * @{
+ */
+#define MSBFIRST        (0)     /**< most significat bit first */
```

Comment:
`LSBFIRST` is not supported in RIOT, as of now. Not defining it will allow users to recognize the presence of an issue at compile time, rather than an assertion blowing up at run time.

I updated the value of MSBFIRST to 1

Reply:
I know. My concern was just that someone could use 1 instead of the symbol.

Reply:
Thanks for pointing that out. With the value of `MSBFIRST` now being set to `1`, this should no longer be an issue. If someone uses `0` instead of `LSBFIRST` this will sadly not be detected at compile time, but the `assert()` is still in place. (Also: As RIOT has (to my very best knowledge) no support for sending the list significant bit of a byte first, there seems to be little to no demand for that.)

Reply:
> As RIOT has (to my very best knowledge) no support for sending the list significant bit of a byte first, there seems to be little to no demand for that.

That's correct. The `assert` is enough to check the condition at execution time.